src/webapp/reports/autogenerated-forms/hooks/useSectionVisibility.ts

```diff
@@ -0,0 +1,121 @@
+import { useMemo } from "react";
+import { Section } from "../../../../domain/common/entities/DataForm";
+import { ToggleDataElement } from "../../../../domain/common/entities/ToggleMultiple";
+import { DataFormInfo } from "../AutogeneratedForm";
+import { evaluateTotalRule, verifyConditionByDataValueType } from "./useApplyRules";
+import { isDataValueEnabled } from "../DataElementItem";
+import { DataTableSectionObj } from "../DataTableSection";
+
+export type SectionVisibilityState = {
+    isDisabled: boolean;
+    isOpen: boolean;
+    isVisible: boolean;
+};
+
+export function useSectionVisibility(section: DataTableSectionObj, dataFormInfo: DataFormInfo): SectionVisibilityState {
+    const { toggle, toggleMultiple } = section;
```

Comment:
Check my comment in `GetDataFormUseCase.ts` just to be sure if we're filtering sections in the right place.